test/TestRist.cpp

```diff
@@ -0,0 +1,329 @@
+        // Does not work with 6 sec or shorter wait
+        bool successfulWait =
+            disconnectCondition.wait_for(lock, std::chrono::seconds(10), [&]() { return disconnected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for sender disconnect";
+    }
+}
+
+// TODO Enable test when STAR-255 is fixed
+TEST(TestRist, DISABLED_ReceiverCloseConnections) {
+    RISTNetReceiver receiver;
+    std::vector<std::string> receiverInterfaces;
+    RISTNetReceiver::RISTNetReceiverSettings receiverSettings;
+    receiverSettings.mPSK = kValidPsk;
+
+    std::condition_variable connectedCondition;
+    std::mutex connectedMutex;
+    bool connected = false;
+    receiver.validateConnectionCallback = [&](const std::string& ipAddress, uint16_t port) {
+        {
+            std::lock_guard<std::mutex> lock(connectedMutex);
+            connected = true;
+        }
+        connectedCondition.notify_one();
+        return std::make_shared<RISTNetReceiver::NetworkConnection>();
+    };
+
+    receiverInterfaces.push_back("rist://@0.0.0.0:8000");
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    std::vector<std::tuple<std::string, int>> senderInterfaces;
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kValidPsk;
+
+    senderInterfaces.push_back(std::tuple<std::string, int>("rist://127.0.0.1:8000", 1));
+
+    // notice when sender disconnects
+    std::condition_variable disconnectCondition;
+    std::mutex disconnectMutex;
+    bool disconnected = false;
+    receiver.clientDisconnectedCallback = [&](const std::shared_ptr<RISTNetReceiver::NetworkConnection>& connection,
+                                              const rist_peer& peer) {
+        {
+            std::lock_guard<std::mutex> lock(disconnectMutex);
+            disconnected = true;
+        }
+        disconnectCondition.notify_one();
+    };
+
+    RISTNetSender sender;
+    ASSERT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+
+    // check for sender connecting
+    {
+        std::unique_lock<std::mutex> lock(connectedMutex);
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(10), [&]() { return connected; });
+        ASSERT_TRUE(successfulWait) << "Timeout waiting for sender to connect";
+    }
+
+    rist_peer* client;
+    receiver.getActiveClients(
+        [&](std::map<rist_peer*, std::shared_ptr<RISTNetReceiver::NetworkConnection>>& activeClients) {
+            EXPECT_EQ(activeClients.size(), 1);
+            client = activeClients.begin()->first;
+        });
+
+    RISTNetSender sender2;
+    ASSERT_TRUE(sender2.initSender(senderInterfaces, senderSettings));
+
+    // check for sender2 connecting
+    {
+        std::unique_lock<std::mutex> lock(connectedMutex);
+        connected = false;
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(10), [&]() { return connected; });
```

Comment:
Maybe these different durations in the lambdas should be named constants to convey the reason for the different timeouts a little bit better...